root/usr/share/fdi/facts/ethtool.rb

```diff
@@ -17,35 +17,37 @@
 
   setcode do
     interface_info = {}
-    Facter.value(:networking)['interfaces'].keys.each do |interface|
-      Facter.debug("Running ethtool on interface #{interface}")
-      output = Facter::Core::Execution.exec("ethtool #{interface} 2>/dev/null")
-      if output.nil?
-        Facter.debug("Execution of ethtool on interface #{interface} failed")
-      else
-        attributes = {}
-        output.each_line do |line|
-          next if line.nil? or line == ""
-          case line.strip
-          when /Speed: (.*)Mb/
-            attributes[:speed] = $1
-          when /Duplex: (.*)/
-            attributes[:duplex] = $1.downcase
-          when /Port: (.*)/
-            attributes[:port] = $1
-          when /Auto-negotiation: (.*)/
-            attributes[:auto_negotiation] = ($1 == 'on').to_s
-          when /^Wake-on: (.*)/
-            attributes[:wol] = $1.include?('g')
-          when /Link detected: (.*)/
-            attributes[:link] = ($1 == 'yes').to_s
+    if Facter.value(:networking) && Facter.value(:networking)['interfaces']
```

Comment:
I would store the fact in a variable given how often you use it
```suggestion
    if (networking = Facter.value(:networking)) && networking['interfaces']
```

Reply:
Well this isn't my code but...